### PR TITLE
 Fix #7560: Use GNSS API if available

### LINKED ISSUE #7560: c:geo shows less satellites
##### Detailed steps causing the problem:
Start c:geo and wait for a fix
Start "gps test" app and wait for a fix
compare the results 

##### Actual behavior after performing these steps:
c:geo shows 19/22 satelites
gps test shows 24/32 satelites

##### Expected behavior after performing these steps:
c:geo also shows 24/32 satelites

##### Version of c:geo used:
installed yesterday via play store

##### Is the problem reproducible for you?
Yes

##### System information:
Using a Huawei Mate 20 Pro, that utilizes gps, glonass, beidou and galileo satelites.

### PR DESCRIPTION
**Pull request by UniQP**
Setting the 'Do not merge' tag until tested properly.

### CODE REVIEW

**Comment by UniQP** [comment]
I tested it and the number of satellites matches the number shown by GPSTest (> 30 with the PR and at most 26 with current nightly).

File: main/src/cgeo/geocaching/MainActivity.java [hunk 1]
```diff
@@ -198,11 +200,13 @@ private static String formatAddress(final Address address) {
...
+    private final Consumer<GnssStatusProvider.Status> satellitesHandler = new Consumer<Status>() {
         @Override
-        public void accept(final Status gpsStatus) {
-            if (gpsStatus.gpsEnabled) {
-                navSatellites.setText(res.getString(R.string.loc_sat) + ": " + gpsStatus.satellitesFixed + '/' + gpsStatus.satellitesVisible);
+        public void accept(final Status gnssStatus) {
+            if (gnssStatus.gnssEnabled) {
+                navSatellites.setText(res.getString(R.string.loc_sat) + ": " + gnssStatus.satellitesFixed + '/' + gnssStatus.satellitesVisible);
+            } else if (VERSION.SDK_INT >= VERSION_CODES.N) {
+                navSatellites.setText(res.getString(R.string.loc_gnss_disabled));
```
Comment (Bananeweizen) [inline main/src/cgeo/geocaching/MainActivity.java]: I'm not sure of this additional new status. I never heard of GNSS before, and I would guess the same for many other of our users. For those "GPS" is what they want, independent of what we use to calculate the status.
Basically, this is the general UX best practice "never use implementation terminology in the user interface, the user has a different mental model than the developer".

File: main/src/cgeo/geocaching/sensors/GnssStatusProvider.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            public void subscribe(final ObservableEmitter<Status> subscriber) throws Exception {
+                final LocationManager geoManager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+                final Callback callback = new Callback() {
+                    Status latest = NO_GNSS;
+
+                    @Override
+                    public void onSatelliteStatusChanged(final GnssStatus status) {
+                        final int visible = status.getSatelliteCount();
+                        int fixed = 0;
+                        for (int satellitIndex = 0; satellitIndex < visible; satellitIndex++) {
```
Comment (Bananeweizen) [inline main/src/cgeo/geocaching/sensors/GnssStatusProvider.java]: satellit_e_Index

File: main/src/cgeo/geocaching/MainActivity.java (hunk 1 above)
Comment (Lineflyer) [reply main/src/cgeo/geocaching/MainActivity.java]: Yes, lets leave the UI untouched.

 Basically all people only know GPS and still expect, that all other sat systems are used as well.

**Pull request merged**